server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -351,8 +410,15 @@ private void consume(QuerySearchResult result, Runnable next) {
         if (hasFailure()) {
             result.consumeAll();
             next.run();
-        } else if (result.isNull()) {
-            result.consumeAll();
+        } else if (result.isNull() || result.isPartiallyReduced()) {
+            if (result.isPartiallyReduced()) {
+                if (result.hasConsumedTopDocs() == false) {
+                    result.consumeTopDocs();
+                }
+                result.releaseAggs();
+            } else {
+                result.consumeAll();
+            }
```

Comment:
why the distinction here between isNull and isPartiallyReduced? consumeAll is very very similar?

Reply:
The problem was with suggest and profiling. Those two things I'd want to still live on since they're not partially reducing. That's where I needed something in between null and a full result.

Reply:
I think this is all dead code now actually ... see https://github.com/elastic/elasticsearch/pull/125299 ... I'll drop this code block here as well.